plugins/de.cognicrypt.staticanalyzer.kotlin/src/de/cognicrypt/staticanalyzer/kotlin/utilities/KotlinUtils.java

```diff
@@ -0,0 +1,261 @@
+				}
+				// because in some projects the class names aren't renamed
+				else
+					srcFilename = classFileName + ".kt";
+
+				for (final IPackageFragment l : JavaCore.create(currentProject).getPackageFragments()) {
+					// this check is needed because IJavaProject.getPackageFragments() returns dependencies as well
+					if(l.getKind() == IPackageFragmentRoot.K_SOURCE) {
+						// removing the <project_name> from path returned by IPackageFragment.getPath() because IProject.getFile() also appends it
+						String[] originalPath = l.getPath().toString().split(File.separator);
+						String[] modifiedPath = Arrays.copyOfRange(originalPath, 2, originalPath.length);
+						String packageName = String.join(File.separator, modifiedPath);
+
+						IFile sourceFile = currentProject.getFile(packageName + File.separator + srcFilename);
+						if(sourceFile.exists())
+							return (IResource) sourceFile;
+					}
+				}
+			}
+		}
+		
+		throw new ClassNotFoundException("Class " + className + " not found.");
+	}
+	
+	public static boolean verifyKotlinDependency(IJavaProject javaProject) {
+ 		IProject p = javaProject.getProject();
+ 		IFile pomFile = p.getFile("pom.xml");
+ 		boolean isPresent = false;
+ 		try {
+ 			MavenProject project = loadProject(pomFile.getLocation().toFile());
+ 			List<Dependency> dependencies = project.getDependencies();
+ 			
+ 			for (Dependency dependency : dependencies) {
+ 				System.out.println(dependency);
+ 				if(dependency.getArtifactId().equals("kotlin-stdlib"))
+ 					isPresent = true;
+ 			}
+ 			if(!isPresent) {
+ 				boolean accepted = requestUsersPermission();
+ 				if(accepted) {
+ 					addKotlinDependency(project, dependencies);
+ 					isPresent = true;
+ 				}
+ 			}
+ 		} catch (Exception e) {
+ 			e.printStackTrace();
+ 		}
+ 		return isPresent;
+ 	}
+	
+	public static void waitForBuildAndRefreshJobs() {
+		ArrayList<Job> jobs = new ArrayList<Job>();
+		while(true)
+		{
+			jobs.clear();
+			jobs.addAll(Arrays.asList(Job.getJobManager().find(ResourcesPlugin.FAMILY_AUTO_BUILD)));
+			jobs.addAll(Arrays.asList(Job.getJobManager().find(ResourcesPlugin.FAMILY_AUTO_REFRESH)));
+			jobs.addAll(Arrays.asList(Job.getJobManager().find(ResourcesPlugin.FAMILY_MANUAL_BUILD)));
+			jobs.addAll(Arrays.asList(Job.getJobManager().find(ResourcesPlugin.FAMILY_MANUAL_REFRESH)));
+			if(jobs.isEmpty()) {
+				System.out.println("No pending jobs found.");
+				return;
+			}
+			System.out.println("Waiting for " + jobs.size() + " Jobs to finish...");
+			
+			boolean buildSuccess;
+			do {
+				buildSuccess = true;
+				for (Job job : jobs) {
+					if(job.getState() != Job.NONE)
+						buildSuccess = false;
+				}
+			}
+			while(!buildSuccess);
+		}
+	}
+
+	private static void addKotlinDependency(MavenProject project, List<Dependency> dependencies) {
+		Dependency kotlinDependency = new Dependency();
+		kotlinDependency.setGroupId("org.jetbrains.kotlin");
+		kotlinDependency.setArtifactId("kotlin-stdlib");
```

Comment:
Is there a way to make this more dynamic? We likely always want the latest version of the Kotlin plugin to be added, not always Kotline 1.3.61, right?